src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -567,4 +571,64 @@ public virtual void PrintMessageNormalized(TextWriter writer, string message, bo
         else if (!removeLastNewLine)
             writer.WriteLine();
     }
+
+    /// <summary>
+    /// Adds a new runtime option. Throws if option already exists.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    public void AddOption<T>(string name, T value)
+    {
+        Options.Add(name, value);
+    }
+
+    /// <summary>
+    /// Trys to add a new runtime option.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    /// <returns>True if successful. False otherwise.</returns>
+    public bool TryAddOption<T>(string name, T value)
```

Comment:
>name [](http://example.com/codeflow?start=39&length=4)

I think it is better to check and throw if null name for better top stack frame.

Reply:
Think I should throw for null name here or just return false?

Reply:
Better to throw to be consistent with the Dictionary implementation. 
```C#
            Dictionary<string, string> o = new();
            o.TryAdd(null!, "");
```

this will produce 

```
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'key')
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
```

Reply:
Fixed to match Dictionary.